server/homebrew.api.js

```diff
@@ -41,154 +41,195 @@ const excludePropsFromUpdate = (brew)=>{
+		});
 
-		obj = obj.toObject();
-		obj.gDrive = false;
-		return res.status(200).send(obj);
-	});
+	saved = saved.toObject();
+	saved.gDrive = false;
+	return saved;
 };
 
-const updateBrew = (req, res)=>{
-	HomebrewModel.get({ editId: req.params.id })
-		.then((brew)=>{
-			const updateBrew = excludePropsFromUpdate(req.body);
-			brew = _.merge(brew, updateBrew);
-			brew.text = mergeBrewText(brew);
-
-			// Compress brew text to binary before saving
-			brew.textBin = zlib.deflateRawSync(brew.text);
-			// Delete the non-binary text field since it's not needed anymore
-			brew.text = undefined;
-			brew.updatedAt = new Date();
-
-			if(req.account) {
-				brew.authors = _.uniq(_.concat(brew.authors, req.account.username));
-			}
-
-			brew.markModified('authors');
-			brew.markModified('systems');
-
-			brew.save((err, obj)=>{
-				if(err) throw err;
-				return res.status(200).send(obj);
-			});
-		})
-		.catch((err)=>{
-			console.error(err);
-			return res.status(500).send('Error while saving');
-		});
+const newGoogleBrew = async (account, brew, res)=>{
+	const oAuth2Client = GoogleActions.authCheck(account, res);
+
+	return await GoogleActions.newGoogleBrew(oAuth2Client, brew);
 };
 
-const deleteBrew = (req, res)=>{
-	HomebrewModel.find({ editId: req.params.id }, (err, objs)=>{
-		if(!objs.length || err) {
-			return res.status(404).send('Can not find homebrew with that id');
-		}
+const newBrew = async (req, res)=>{
+	const brew = req.body;
+	const { transferToGoogle } = req.query;
 
-		const brew = objs[0];
+	delete brew.editId;
+	delete brew.shareId;
+	delete brew.googleId;
 
-		if(req.account) {
-			// Remove current user as author
-			brew.authors = _.pull(brew.authors, req.account.username);
-			brew.markModified('authors');
-		}
+	beforeNewSave(req.account, brew);
 
-		if(brew.authors.length === 0) {
-			// Delete brew if there are no authors left
-			brew.remove((err)=>{
-				if(err) return res.status(500).send('Error while removing');
-				return res.status(200).send();
+	let saved;
+	if(transferToGoogle) {
+		saved = await newGoogleBrew(req.account, brew, res)
+			.catch((err)=>{
+				res.status(err.status || err.response.status).send(err.message || err);
 			});
-		} else {
-			// Otherwise, save the brew with updated author list
-			brew.save((err, savedBrew)=>{
-				if(err) throw err;
-				return res.status(200).send(savedBrew);
+	} else {
+		saved = await newLocalBrew(brew)
+			.catch((err)=>{
+				res.status(500).send(err);
 			});
-		}
-	});
+	}
+	if(!saved) return;
+	return res.status(200).send(saved);
 };
 
-const newGoogleBrew = async (req, res, next)=>{
-	let oAuth2Client;
+const updateBrew = async (req, res)=>{
+	let brew = excludePropsFromUpdate(req.body);
+	const { transferToGoogle, transferFromGoogle } = req.query;
 
-	try {	oAuth2Client = GoogleActions.authCheck(req.account, res); } catch (err) { return res.status(err.status).send(err.message); }
+	let saved;
+	if(!!brew.googleId && transferFromGoogle) {
```

Comment:
Excuse me for my ignorance, but what does `!!brew.googleId` mean in JavaScript? I understand what it means in C, but not in JS.

My concern is that if it is some kind of a tricky rare construct, then it might be better to refactor it to make the code more readable

Reply:
Doesn't Javascript already implicitly cast all variables to booleans when evaluated in an `if` statement? I.e,. `if(brew.googleId)` would evaluate false for all of those examples you listed anyway.

Reply:
Removed both uses of the `!!`